docs/root/configuration/http/http_conn_man/headers.rst

```diff
@@ -9,6 +9,25 @@ is being received) as well as during encoding (when the response is being sent).
 .. contents::
   :local:
 
+.. _config_http_conn_man_headers_scheme:
+
+:scheme
+-------
+
+Envoy will always set the *:scheme* header while processing a request. It should always be available to filters, and should be forwarded upstream for HTTP/2 and HTTP/3, where :ref:`config_http_conn_man_headers_x-forwarded-proto` will be sent for HTTP/1.1.
+
+For HTTP/2, and HTTP/3, incoming *:scheme* headers are trusted and propogated through upstream.
+For HTTP/1, the *:scheme* header will be set
+1) From the absolute URL if present and valid. An invalid (not "http" or "https") scheme, or an https scheme over an unencrypted connection will result in Envoy rejecting the request. This is the only scheme validation Envoy performs as it avoids a HTTP/1.1-specific privledge escalation attack for edge Envoys [1]_ which doesn't have a comparable vector for HTTP/2 and above [2]_.
+2) From the value of the :ref:`config_http_conn_man_headers_x-forwarded-proto` header after sanitization (to valid *x-forwarded-proto* from trusted downstreams, otherwise based on downstream encryption level).
```

Comment:
please use double-backtick literal for ` ``x-forward-proto`` `

Reply:
I think I'd prefer a follow-up sweep to tthe file at large. I think all other headers in this file are *d rather than ``d and I'd prefer consistency
*x-b3-traceid* *x-b3-spanid* *x-b3-parentspanid*  *x-b3-sampled* *x-datadog-trace-id* *x-envoy-external-address* etc

Reply:
k, thanks